File: reports/Brych_Maxim/lab3/src/Triangle.java
public class Triangle {
    Float cathet_1;
    Float cathet_2;
    Float hypotenuse;

    public Triangle(Float cathet_1, Float cathet_2, Float hypotenuse) {
        this.cathet_1 = cathet_1;
        this.cathet_2 = cathet_2;
        this.hypotenuse = hypotenuse;
    }

    public void setTriangle(Float cathet_1, Float cathet_2, Float hypotenuse) {
        this.cathet_1 = cathet_1;
        this.cathet_2 = cathet_2;
        this.hypotenuse = hypotenuse;
    }

    public Triangle() {
        this.cathet_1 = null;
        this.cathet_2 = null;
        this.hypotenuse = null;
    }

    public double Square()
    {

        return Math.sqrt(Perimeter()*(Perimeter()-cathet_1)*(Perimeter()-cathet_2)*(Perimeter()-hypotenuse));
    }

    public Float Perimeter()
    {
        return cathet_1 + cathet_2 + hypotenuse;
    }

    public boolean isRectangular()
    {
        return Math.pow(cathet_1, 2) + Math.pow(cathet_2, 2)
                        == Math.pow(hypotenuse, 2);
    }

    public Float getCathet_1() {
        return cathet_1;
    }

    public void setCathet_1(Float cathet_1) {
        this.cathet_1 = cathet_1;
    }

    public Float getCathet_2() {
        return cathet_2;
    }

    public void setCathet_2(Float cathet_2) {
        this.cathet_2 = cathet_2;
    }

    public Float getHypotenuse() {
        return hypotenuse;
    }

    public void setHypotenuse(Float hypotenuse) {
        this.hypotenuse = hypotenuse;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Triangle triangle = (Triangle) o;
        return cathet_1.equals(triangle.cathet_1) && cathet_2.equals(triangle.cathet_2) && hypotenuse.equals(triangle.hypotenuse);
    }

    @Override
    public String toString() {

        return "Катет №1 = " + cathet_1 +
                ", Катет №2 = " + cathet_2 +
                ", Гипотенуза = " + hypotenuse;
    }
}

File: Reports/Vasilyuk_Pavel/lab3/src/Triangle.kt
package org.example.main

import kotlin.math.pow

public class Triangle {
    private var side1: Double = 0.0
    private var side2: Double = 0.0
    private var side3: Double = 0.0


    constructor(side_1: Double, side_2: Double, side_3: Double)
    {
        this.side1 = side_1
        this.side2 = side_2
        this.side3 = side_3
    }

    constructor()
    {
        this.side1 = 0.0
        this.side2 = 0.0
        this.side3 = 0.0
    }

    public fun setTriangle(side_1: Double, side_2: Double, side_3: Double)
    {
        this.side1 = side_1
        this.side2 = side_2
        this.side3 = side_3
    }

    public fun square(): Double
    {
        return (3.0.pow(1.0/2.0) / 4 * side1.pow(2.0))
    }

    public fun perimeter(): Double
    {
        return side1 * 3
    }

    public fun isRectangular(): Boolean {
        return (side1 == side2 && side2 == side3)
    }

    var Side1: Double?
        get() = side1;
        set(value) {
            if (value != null) {
                side1 = value
            }
        }

    var Side2: Double?
        get() = side2;
        set(value) {
            if (value != null) {
                side2 = value
            }
        }

    var Side3: Double?
        get() = side3;
        set(value) {
            if (value != null) {
                side3 = value
            }
        }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val triangle: Triangle = o as Triangle
        return side1.equals(triangle.side1) && side2.equals(triangle.side2) && side3.equals(triangle.side3)
    }

    override fun toString(): String
    {
        return "sides = $side1, $side2, $side3"
    }
}